src/routes/signup/webauthn.ts

```diff
@@ -0,0 +1,163 @@
+import { sendError } from '@/errors';
+import { UserRegistrationOptions } from '@/types';
+import { ENV, getSignInResponse, getUserByEmail, gqlSdk } from '@/utils';
+import { RequestHandler } from 'express';
+
+import {
+  generateRegistrationOptions,
+  verifyRegistrationResponse,
+} from '@simplewebauthn/server';
+import { email, Joi, passwordInsert, registrationOptions } from '@/validation';
+import { createUserAndSendVerificationEmail } from '@/utils/user/email-verification';
+
+export const signUpWebauthnSchema = Joi.object({
+  email: email.required(),
+  password: passwordInsert.optional(),
```

Comment:
why adding password as an option? It is not required by webauthn (and we can also assume users opting for wbeauthn don't want any password).

They can eventually add a password after sign-up

Reply:
I was wondering if this is a nice optional feature or should be removed, so you propose not to have it?

Reply:
I think that is better not to have it. Users can set their password at a later stage in using the `/user/password` endpoint.
We are doing that way for Oauth or passwordless email, I would keep the same logic

Reply:
Changed.